zerver/tests/test_message_send.py

```diff
@@ -2362,6 +2365,86 @@ def test_personal(self) -> None:
             receiver=self.example_user("othello"),
         )
 
+    def test_direct_message_initiator_group_setting(self) -> None:
+        """
+        Tests that direct_message_initiator_group_setting works correctly.
+        """
+        user_profile = self.example_user("hamlet")
+        polonius = self.example_user("polonius")
+        admin = self.example_user("iago")
+        realm = user_profile.realm
+        huddle = [user_profile, admin, polonius]
+        administrators_system_group = NamedUserGroup.objects.get(
+            name=SystemGroups.ADMINISTRATORS, realm=realm, is_system_group=True
+        )
+        self.login_user(user_profile)
+        self.send_personal_message(user_profile, polonius)
+        do_change_realm_permission_group_setting(
+            realm,
+            "direct_message_initiator_group",
+            administrators_system_group,
+            acting_user=None,
+        )
+
+        with self.assertRaises(JsonableError):
+            self.send_personal_message(user_profile, self.example_user("cordelia"))
+        with self.assertRaises(JsonableError):
+            self.send_personal_message(user_profile, admin)
+        with self.assertRaises(JsonableError):
+            self.send_huddle_message(user_profile, huddle)
+
+        self.send_personal_message(user_profile, polonius)
```

Comment:
Comment your tests! This line could have before it:

> # We can send to Polonius because we'd previously messaged him.

And then the next block could be:

> # Have the administrator send a message, and verify that allows the user to reply.